p/riscv_test.h

```diff
@@ -190,6 +190,7 @@ handle_exception:                                                       \
   1:    ori TESTNUM, TESTNUM, 1337;                                     \
   write_tohost:                                                         \
         sw TESTNUM, tohost, t5;                                         \
+        sw zero, tohost + 4, t5;                                        \
```

Comment:
I'm confused as to what this is supposed to accomplish.  Is the problem that QEMU is only actioning side effects when both halves of the register have been written?

Reply:
Yes, QEMU set allow_tohost when the low half is written and handle tohost_write when the high half is written:
```

#define TOHOST_OFFSET1 (htifstate->tohost_offset)
#define TOHOST_OFFSET2 (htifstate->tohost_offset + 4)
...

static void htif_mm_write(void *opaque, hwaddr addr,
                            uint64_t value, unsigned size)
{
    HTIFState *htifstate = opaque;
    if (addr == TOHOST_OFFSET1) {
        if (htifstate->env->mtohost == 0x0) {
            htifstate->allow_tohost = 1;
            htifstate->env->mtohost = value & 0xFFFFFFFF;
        } else {
            htifstate->allow_tohost = 0;
        }
    } else if (addr == TOHOST_OFFSET2) {
        if (htifstate->allow_tohost) {
            htifstate->env->mtohost |= value << 32;
            htif_handle_tohost_write(htifstate, htifstate->env->mtohost);
        }
    }
    ...
}
```

Reply:
tbh this seems like a workaround for a QEMU bug.

Reply:
I don't find the spec for htif. No matter for qemu and for spike, tohost itself is  8 bytes. What is different is that
   spike make it just like a shared memory between core and htif. No matter the core writes 4 bytes or 8 bytes. the htif itself read 8 bytes. 
   QEMU make it as a device register. the core should write the whole register to make it work.

Reply:
I know "the spec is the implementation" isn't satisfying, but the fact is, Spike is authoritative on this point.  Nevertheless, your proposed change is harmless, so we can just run with it for now.

Reply:
OK, Thanks.